### ISSUE TITLE
Life is Strange: True Colors (936790)

### ISSUE THREAD

**Issue by headlesscyborg** [ValveSoftware/Proton#5147]
# Compatibility Report
- Name of the game with compatibility issues: Life is Strange: True Colors
- Steam AppID of the game: 936790

## System Information
- GPU: GTX 1660 Ti
- Driver/LLVM version: nvidia 470.63.01-1
- Kernel version: 5.13.13-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/headlesscyborg/20e8668ce792f0ee065871584de9d497
- Proton version: 6.3-6, Experimantal (9th September 2021)

## I confirm:
- [ X] that I haven't found an existing compatibility report for this game.
- [ X] that I have checked whether there are updates for my system available.

[steam-936790.log](https://github.com/ValveSoftware/Proton/files/7138366/steam-936790.log)

## Symptoms 
The game starts fine but a fatal error appears soon, the game keeps running but crashes within a minute. I've tried both Proton 6.3.6 and Experimental, both DX11 and DX12 renderers and it's exactly the same.
![Snímek z 2021-09-09 18-50-45](https://user-images.githubusercontent.com/28152817/132728940-b3d98f89-caf7-4841-be00-f787b6779a15.png)

## Reproduction
Start the game, browse main menu.

**Comment by JonathanBrouwer** [comment]
Can confirm using Proton Experimental and Proton-GE.

**Comment by kisak-valve** [comment]
Hello @headlesscyborg, `Could not connect. Error code = 0x8004100e` in your log looks like it's close there where the game starts having trouble.

**Comment by Sojiro84** [comment]
Same for me. Seems it might have to do with the controller? Since at the end in the terminal I see:

_Couldn't initialize virtual gamepad: Couldn't open /dev/uinput for writing_

In any case, here is my log: [steam-936790.log](https://github.com/ValveSoftware/Proton/files/7138506/steam-936790.log)

I used Proton Experimental and DX11.

**Comment by ghost** [comment]
0x8004100e is WBEM_E_INVALID_NAMESPACE, a log with WINEDEBUG=+wbemprox might be useful.

**Comment by JonathanBrouwer** [comment]
Ok, this is with `WINEDEBUG=+wbemprox PROTON_LOG=1 %command%`
I'm not really sure what the command you gave does, so let me know if I need to do something else
https://gist.github.com/JonathanBrouwer/022ce3f706ddd6c1bde22a85f57cc621

**Comment by Sojiro84** [comment]
Here you go as well, a log with "WINEDEBUG=+wbemprox": [steam-936790.log](https://github.com/ValveSoftware/Proton/files/7138551/steam-936790.log)

**Comment by ben-romer** [comment]
I've been experimenting with the game and I found that if you start the game without a network connection, the error does not occur.

**Comment by Sojiro84** [comment]
> I've been experimenting with the game and I found that if you start the game without a network connection, the error does not occur.

I was just about to test this, since it seems the issue is network related. Nice!

**Comment by rkfg** [comment]
Maybe it's possible to block networking for this game only using AppArmor? Does the game actually work with no network access, past the main menu?

**Comment by irgendwr** [comment]
**Update: use the latest Proton experimental version instead.**

The workaround below is not required anymore.
If you previously used the workaround, please remove it to allow the online functionality of the game.

---

I could get it to work by blocking access to `see-siren.os.eidos.com` and `see-siren-prod.os.eidos.com`.

You can do this for example by adding
```
0.0.0.0 see-siren.os.eidos.com
0.0.0.0 see-siren-prod.os.eidos.com
```
to your `/etc/hosts`.

Warning: this will disable online functionality of the game (eg. choice stats).

**Comment by kevinlekiller** [comment]
Another solution I found is using : https://github.com/JsBergbau/BindToInterface

I use Steam from Flatpak so I set the paths accordingly:

    cd ~/.var/app/com.valvesoftware.Steam
    wget https://github.com/JsBergbau/BindToInterface/raw/main/bindToInterface.c
    gcc -nostartfiles -fpic -shared bindToInterface.c -o bindToInterface.so -ldl -D_GNU_SOURCE

Then I set the launch properties, using the loopback device as the interface:

`BIND_INTERFACE=lo LD_PRELOAD="$HOME/.var/app/com.valvesoftware.Steam/bindToInterface.so:$LD_PRELOAD" %command%`

**Comment by kevinlekiller** [comment]
Setting the Shadows from Very High to High drops the GPU (6800 XT) power usage from 300 watts to 100 watts and GPU usage from 99% to 11% on the start screen at 3840x2160 resolution on the DX11 version, haven't tried the DX12 version. Lowering Effects Quality from Very High to High is the bext big thing to improve performance after Shadows.

That's with Motion blur / Chromatic Aberration / Film Grain / Vignette / Depth of Field disabled.

You can disable them in `.local/share/Steam/steamapps/compatdata/936790/pfx/drive_c/users/steamuser/AppData/Local/Siren/Saved/Config/WindowsNoEditor/Engine.ini` 

Add  `[SystemSettings]` to the file and look at https://github.com/kevinlekiller/config_files/tree/main/unreal_engine_4 for explanations of how to disable the various settings.

**Comment by kevinlekiller** [comment]
Outside of Big Picture Mode, the game is not detecting controllers, launching the game through Big Picture Mode allows controllers to be detected by the game, tested on Proton Experimental, 6.3 and latest Proton-GE with Dualshock 4 & Steam Controller.

Edit: Works now outside of BPM, must be something triggered by launching in BPM? I did try rebooting / deleting the WINEPREFIX before trying BPM and that didn't work.

**Comment by Sojiro84** [comment]
> Outside of Big Picture Mode, the game is not detecting controllers, launching the game through Big Picture Mode allows controllers to be detected by the game, tested on Proton Experimental, 6.3 and latest Proton-GE with Dualshock 4 & Steam Controller.
> 
> Edit: Works now outside of BPM, must be something triggered by launching in BPM? I did try rebooting / deleting the WINEPREFIX before trying BPM and that didn't work.

Need to test this. I also couldn't get my controller working with the game even though I did saw a po-up from Steam saying a controller config was loaded (even though I don't use custom configs.).

**Comment by samip5** [comment]
> I could get it to work by blocking access to `see-siren.os.eidos.com` and `see-siren-prod.os.eidos.com`.
> 
> You can do this for example by adding
> 
> ```
> 0.0.0.0 see-siren.os.eidos.com
> 0.0.0.0 see-siren-prod.os.eidos.com
> ```
> 
> to your `/etc/hosts`.

Verified that this works on my system too.

**Comment by Sojiro84** [comment]
I can confirm that my xbox one controller works when launching the game with BPM.

Also added

```
0.0.0.0 see-siren.os.eidos.com
0.0.0.0 see-siren-prod.os.eidos.com
```

To my pi-hole and this works as well.

**Comment by ben-romer** [comment]
Steam Controller worked fine here without BPM. Confirming also the /etc/hosts workaround for the networking problem.

Perhaps the crash here is similar or the same cause as the issue Death Stranding is having with connecting to its servers - DS was functional up until a month or so ago and then stopped being able to connect, and can't connect at all right now with the current proton experimental release.

**Comment by ghost** [comment]
I made a simple [patch](https://gist.github.com/EsdrasTarsis/663f35d31645015fec43e8a8fa497cf2) on top of experimental to accept the microsoft/windows/storage namespace in wbemprox, I don't know if this is the best approach to deal with unsupported namespaces, but it should solve the error message issue.

**Comment by ghost** [comment]
I also have this bug with FreeBSD and thought i was alone with it (in fear that could be OS specific), thanks for your investigations!

**Comment by Sojiro84** [comment]
> I made a simple [patch](https://gist.github.com/EsdrasTarsis/663f35d31645015fec43e8a8fa497cf2) on top of experimental to accept the microsoft/windows/storage namespace in wbemprox, I don't know if this is the best approach to deal with unsupported namespaces, but it should solve the error message issue.

Does this mean it is fixed with current Proton Experimental?

**Comment by alpacalife** [comment]
Blocking these worked for me too: `see-siren.os.eidos.com` and `see-siren-prod.os.eidos.com`.

I tested Proton 6.3-6 w/ non directx 12.

Xbox One Controller did not work with Proton 6.3-6. Worked on Proton 6.16-GE-1.

**Comment by Kepsz** [comment]
> Blocking these worked for me too: `see-siren.os.eidos.com` and `see-siren-prod.os.eidos.com`.

Yeah, but the game will not be able to compare your choices with the rest of the world at the end of each episode. So this is only a quick fix, not the solution.

**Comment by ghost** [comment]
> Does this mean it is fixed with current Proton Experimental?

No, I meant that my patch should be applied over the Wine source code of Proton Experimental.

**Comment by Sojiro84** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/5147#issuecomment-917616297

Ahh, that's a shame for me. Guess I will have to wait for it to be fixed in Experimental or maybe Proton-GE to be able to also play my DLC content since somehow the game is programmed in such a way that I need to be able to talk to Eidos in order to see and be able to play the Wavelengths DLC.

**Comment by Kepsz** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/5147#issuecomment-918207174

Wavelengths DLC will only be released at 2021.09.20. The Wavelengths button in the game is just for marketing the DLC, that's why it needs internet.

**Comment by ben-romer** [comment]
> I made a simple [patch](https://gist.github.com/EsdrasTarsis/663f35d31645015fec43e8a8fa497cf2) on top of experimental to accept the microsoft/windows/storage namespace in wbemprox, I don't know if this is the best approach to deal with unsupported namespaces, but it should solve the error message issue.

I've built this patch into a test release on top of 3083b57f002a84b (proton 6.3 current Wine HEAD) and it appears that it delays the error box from appearing until after you quit the game, but it still happens, so we're not quite there yet. Post-patch log attached.

[steam-936790.log](https://github.com/ValveSoftware/Proton/files/7156300/steam-936790.log)

**Comment by igor725** [comment]
I have similar issue, but I can't even see the main menu. Game shows Square Enix logo, then crashes with "Fatal error" message within 15-20sec. I tried to turn off the Internet connection, but it didn't solve the problem.

### System info
GPU: Sapphire RX-580 8GB
CPU: AMD FX-8300
Driver: Mesa 21.2.1
Kernel: 5.14.3-arch1-1
Proton version: 6.3-6

[steam-936790.log](https://github.com/ValveSoftware/Proton/files/7172338/steam-936790.log)

**Comment by ben-romer** [comment]
Looks like today's Proton experimental update fixes the "Fatal Error" message on my system.

**Comment by JonathanBrouwer** [comment]
Can confirm, works perfectly for me now.

**Comment by qptain-Nemo** [comment]
Can also confirm.

**Comment by Witteborn** [comment]
> I could get it to work by blocking access to `see-siren.os.eidos.com` and `see-siren-prod.os.eidos.com`.
> 
> You can do this for example by adding
> 
> ```
> 0.0.0.0 see-siren.os.eidos.com
> 0.0.0.0 see-siren-prod.os.eidos.com
> ```
> 
> to your `/etc/hosts`.

Adding this helps for the main story, but you still can not play the DLC.
The game is thinking you don't own the DLC even though you bought it because it cannot connect to the internet.

**Comment by Kepsz** [comment]
Can also confirm, the game works with proton experimental!

**Comment by YamiYukiSenpai** [comment]
There's this weird glitch where a black screen occurs in this scenario:

<details><summary>Spoiler</summary> 
in the 2nd chapter, just after Gabe's wake, when I confront Mac about Gabe's call, just after he runs away
</details> 

This is my 2nd playthrough with True Colors, and it didn't happen on my first run.

**Comment by htbrown** [comment]
> There's this weird glitch where a black screen occurs in this scenario:

I hate to bring up an old-ish thread, but this happens to me too and I can't figure out a solution. If anyone has, help would be appreciated.

**Comment by Bapf** [comment]
> > There's this weird glitch where a black screen occurs in this scenario:
> 
> I hate to bring up an old-ish thread, but this happens to me too and I can't figure out a solution. If anyone has, help would be appreciated.

Try using Proton 6.3-8.
